### DISCUSSION TITLE
What happen to spawn_blocking when max_blocking_threads is reached

### DISCUSSION THREAD

**Discussion by ririsoft** [tokio-rs/tokio#3848] [Q&A] [answered]
Hello,

I am wondering what happen to [spawn_blocking](https://docs.rs/tokio/1.6.1/tokio/task/fn.spawn_blocking.html) when [max_blocking_threads ](https://docs.rs/tokio/1.6.1/tokio/runtime/struct.Builder.html#method.max_blocking_threads) is reached.

[Current documentation](https://docs.rs/tokio/1.6.1/tokio/task/fn.spawn_blocking.html) is vague about this: 

```
Tokio will spawn more blocking threads when they are requested through this function until the upper limit configured on the Builder is reached. This limit is very large by default, because spawn_blocking is often used for various kinds of IO operations that cannot be performed asynchronously. When you run CPU-bound code using spawn_blocking, you should keep this large upper limit in mind. When running many CPU-bound computations, a semaphore or some other synchronization primitive should be used to limit the number of computation executed in parallel.
```

My understanding from [the code](https://github.com/tokio-rs/tokio/blob/tokio-1.6.1/tokio/src/runtime/blocking/pool.rs#L188) is that it returns quickly without doing anything. Am I right ? Is that the behavior one would expect ? Should an issue be opened to discuss this further, at least to improve the documentation ?

Thanks in advance for your comments here.

**Comment by Darksonn** [comment, marked as answer]
The task is put in a queue, and it will start running once one of the blocking tasks have finished, making up space in the pool.
